src/pepr/operator/controllers/network/generators/kubeNodes.ts

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright 2024 Defense Unicorns
+ * SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
+ */
+
+import { KubernetesListObject } from "@kubernetes/client-node";
+import { V1NetworkPolicyPeer, V1NodeCondition, V1NodeAddress } from "@kubernetes/client-node";
+import { K8s, kind } from "pepr";
+import { Component, setupLogger } from "../../../../logger";
+import { RemoteGenerated } from "../../../crd";
+import { anywhere } from "./anywhere";
+import { UDSConfig } from "../../../../config";
+
+const log = setupLogger(Component.OPERATOR_GENERATORS);
+
+// Maintain a set of all node internal IPs
+const nodeSet = new Set<string>();
+
+/**
+ * Initialize the node targets by fetching the current nodes in the cluster
+ * and populating the nodeSet with their Internal IPs.
+ */
+export async function initAllNodesTarget() {
+  try {
+    const nodes = await fetchKubernetesNodes();
+    nodeSet.clear();
+
+    for (const node of nodes.items) {
+      const ip = getNodeInternalIP(node);
+      if (ip) nodeSet.add(ip);
+    }
+  } catch (err) {
+    log.error("error fetching node IPs:", err);
+  }
+}
+
+/**
+ * Returns the egress CIDRs of all known nodes as network policy peers.
+ * If none are known, defaults to 0.0.0.0/0 and logs a warning.
+ */
+export function nodeCIDRs(): V1NetworkPolicyPeer[] {
+  const policies = buildNodePolicies([...nodeSet]);
+  if (policies.length > 0) return policies;
+
+  log.warn("Unable to get API server CIDR, defaulting to 0.0.0.0/0");
+  return [anywhere];
+}
+
+/**
+ * When a node is created or updated, if it's Ready, add its IP to the set,
+ * rebuild the policies, and update the NetworkPolicies.
+ */
+export async function updateKubeNodesFromCreateUpdate(node: kind.Node) {
+  const isReady = node.status?.conditions?.some(
+    (condition: V1NodeCondition) => condition.type === "Ready" && condition.status === "True",
+  );
+
+  if (isReady) {
+    const ip = getNodeInternalIP(node);
+    if (ip) nodeSet.add(ip);
+  }
+
+  const policies = buildNodePolicies([...nodeSet]);
+  await updateKubeNodesNetworkPolicies(policies);
+}
+
+/**
+ * When a node is deleted, remove its IP from the set, rebuild the policies,
+ * and update the NetworkPolicies.
+ */
+export async function updateKubeNodesFromDelete(node: kind.Node) {
+  const ip = getNodeInternalIP(node);
+  if (ip) nodeSet.delete(ip);
+
+  const policies = buildNodePolicies([...nodeSet]);
+  await updateKubeNodesNetworkPolicies(policies);
+}
+
+/**
+ * Fetch all Kubernetes nodes.
+ */
+async function fetchKubernetesNodes(): Promise<KubernetesListObject<kind.Node>> {
+  return K8s(kind.Node).Get();
+}
+
+/**
+ * Update all NetworkPolicies labeled with uds/generated=KubeNodes to
+ * reflect the given node CIDRs.
+ */
+export async function updateKubeNodesNetworkPolicies(newNodes: V1NetworkPolicyPeer[]) {
+  const netPols = await K8s(kind.NetworkPolicy)
+    .WithLabel("uds/generated", RemoteGenerated.KubeNodes)
+    .Get();
```

Comment:
Mostly just a comment because this line was extremely relevant and problematic in the past - the label we are selecting on gets added to our policies [here](https://github.com/defenseunicorns/uds-core/blob/970-all-nodes/src/pepr/operator/controllers/network/generate.ts#L101-L104). We could update that comment line to indicate that it's also use for the `KubeNodes` policies.

Reply:
👍 added in 024b5874